### DISCUSSION TITLE
How to prevent form rerender when Input component inside FormControl changing?

### DISCUSSION THREAD

**Discussion by spxf** [shadcn-ui/ui#4755] [Q&A]
When using form component, i notice that when i change input, **sometimes** component rerendered especially when deleting all, type the first character or input UTF-8 text.  Am i missing something?
Reproduced link:  [https://codesandbox.io/p/devbox/empty-paper-wklktn?file=%2Fsrc%2Fpages%2FTest.tsx%3A1%2C1](url)

**Comment by abel-officener** [comment]
shadcn is using useFormContext of react-hook-form.
As a characteristic of Context, it uses the value called from the same Context, and if the value is modified in the same Context, rendering occurs.

shadcn form.tsx : 


```
const useFormField = () => {
  const fieldContext = React.useContext(FormFieldContext);
  const itemContext = React.useContext(FormItemContext);
  const { getFieldState, formState } = useFormContext();

  const fieldState = getFieldState(fieldContext.name, formState);

  if (!fieldContext) {
    throw new Error('useFormField should be used within <FormField>');
  }

  const { id } = itemContext;

  return {
    id,
    name: fieldContext.name,
    formItemId: `${id}-form-item`,
    formDescriptionId: `${id}-form-item-description`,
    formMessageId: `${id}-form-item-message`,
    ...fieldState,
  };
};
```
